pkg/teleporter/relayer.go

```diff
@@ -0,0 +1,377 @@
+	relayerAddress string,
+	relayerPrivateKey string,
+	network models.Network,
+	subnetID string,
+	blockchainID string,
+	teleporterContractAddress string,
+	teleporterRegistryAddress string,
+) error {
+	awmRelayerConfig := config.Config{}
+	if utils.FileExists(relayerConfigPath) {
+		bs, err := os.ReadFile(relayerConfigPath)
+		if err != nil {
+			return err
+		}
+		if err := json.Unmarshal(bs, &awmRelayerConfig); err != nil {
+			return err
+		}
+	} else {
+		awmRelayerConfig = createRelayerConfig(
+			logging.Info.LowerString(),
+			relayerStorageDir,
+			network.ID,
+			network.Endpoint,
+		)
+	}
+	host, port, err := getURIHostAndPort(network.Endpoint)
+	if err != nil {
+		return err
+	}
+	addChainToRelayerConfig(
+		&awmRelayerConfig,
+		host,
+		port,
+		subnetID,
+		blockchainID,
+		teleporterContractAddress,
+		teleporterRegistryAddress,
+		relayerAddress,
+		relayerPrivateKey,
+	)
+	bs, err := json.MarshalIndent(awmRelayerConfig, "", "  ")
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(relayerConfigPath, bs, constants.WriteReadReadPerms); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createRelayerConfig(
+	logLevel string,
+	storageLocation string,
+	networkID uint32,
+	endpoint string,
+) config.Config {
+	return config.Config{
+		LogLevel:            logLevel,
+		NetworkID:           networkID,
+		PChainAPIURL:        endpoint,
+		EncryptConnection:   false,
+		StorageLocation:     storageLocation,
+		ProcessMissedBlocks: false,
+		SourceSubnets:       []*config.SourceSubnet{},
+		DestinationSubnets:  []*config.DestinationSubnet{},
+	}
+}
+
+func addChainToRelayerConfig(
+	relayerConfig *config.Config,
+	host string,
+	port uint32,
+	subnetID string,
+	blockchainID string,
+	teleporterContractAddress string,
+	teleporterRegistryAddress string,
+	relayerRewardAddress string,
+	relayerFundedAddressKey string,
+) {
+	source := &config.SourceSubnet{
+		SubnetID:          subnetID,
+		BlockchainID:      blockchainID,
+		VM:                config.EVM.String(),
+		EncryptConnection: false,
+		APINodeHost:       host,
+		APINodePort:       port,
+		MessageContracts: map[string]config.MessageProtocolConfig{
+			teleporterContractAddress: {
+				MessageFormat: config.TELEPORTER.String(),
+				Settings: map[string]interface{}{
+					"reward-address": relayerRewardAddress,
+				},
+			},
+			offchainregistry.OffChainRegistrySourceAddress.Hex(): {
+				MessageFormat: config.OFF_CHAIN_REGISTRY.String(),
+				Settings: map[string]interface{}{
+					"teleporter-registry-address": teleporterRegistryAddress,
+				},
+			},
+		},
+	}
+	destination := &config.DestinationSubnet{
+		SubnetID:          subnetID,
+		BlockchainID:      blockchainID,
+		VM:                config.EVM.String(),
+		EncryptConnection: false,
+		APINodeHost:       host,
+		APINodePort:       port,
+		AccountPrivateKey: relayerFundedAddressKey,
+	}
+	sources := relayerConfig.SourceSubnets
+	found := false
+	for _, s := range sources {
```

Comment:
nit: replace with utils.Filter and check len(filteredList)>0

Reply:
do not agree to create a list to check if a element satisfy a condition, probably we can add a generic function to search for a condition in a list